src/main/java/com/google/api/codegen/DiscoveryFragmentGeneratorApi.java

```diff
@@ -115,6 +126,28 @@ protected void process() throws Exception {
     }
   }
 
+  /*
+   * Parses strings of the format
+   * "go:https://www.hello.com,java:https://www.world.com" and returns the value
+   * of the key corresponding to id.
+   */
+  private String parseAuthUrl(String authUrl, String id) {
```

Comment:
Would it be conceptually simpler to parse this once into a language:url map, and then just get values from the map?